cmd/dep/ensure_test.go

```diff
@@ -186,3 +186,102 @@ func TestDeduceConstraint(t *testing.T) {
 		}
 	}
 }
+
+func TestEnsureUpdate(t *testing.T) {
+	test.NeedsExternalNetwork(t)
+	test.NeedsGit(t)
+
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+
+	h.TempDir("src")
+	h.Setenv("GOPATH", h.Path("."))
+
+	m := `package main
+
+import (
```

Comment:
@tro3 I was able to get the update test (not the others in `ensure_test.go`) working with golden files. Do you want me to hold off on that or finish up the rest of the file?